R/simulate_tools.R

```diff
@@ -409,15 +418,17 @@ spim_simulate_process_output <- function(obj, combined_region, regions,
   ret$multivariant_Rt_general <- NULL
   ret$multivariant_eff_Rt_general <- NULL
 
-  f <- function(x) x[, , output_region, , drop = FALSE]
+  if (output_region_only) {
```

Comment:
If output_region ever non-NULL while this is `FALSE`, or this `TRUE` while output_region is `NULL`? 

If not then remove output_region_only as an argument and use

```
if (!is.null(output_region)) {
...
}
```

If this is not sufficient can you leave a comment explaining why?

Reply:
Problem is we always declare a value for `output_region` in line 415 above. So this is always non-NULL whilst `outputs_region_only` can be FALSE (e.g. as it's currently the case in the simulation of MTPs). Alternatively, I could declare another variable in line 415 and then use what you propose?

Reply:
I would write something like:

```
  output_region_only <- !is.null(output_region)
  output_region <- output_region %||% combined_region
```

above so that it is not possible to use the function with incompatible arguments

Reply:
(you might also add a comment above this bit of code explaining the above too so it's less mysterious to people like me)

Reply:
Ah yeah, nice one! Change pushed